sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -197,4 +218,21 @@ private ArrayList<TokenCredential> getCredentialsChain() {
         output.add(new AzurePowerShellCredential(tenantId, identityClientOptions.clone()));
         return output;
     }
+
+    private WorkloadIdentityCredential getWorkloadIdentityCredentialIfAvailable() {
+        Configuration configuration = identityClientOptions.getConfiguration() == null
+            ? Configuration.getGlobalConfiguration().clone() : identityClientOptions.getConfiguration();
+
+        String tenantId = configuration.get(Configuration.PROPERTY_AZURE_TENANT_ID);
+        String federatedTokenFilePath = configuration.get(AZURE_FEDERATED_TOKEN_FILE);
+        String azureAuthorityHost = configuration.get(Configuration.PROPERTY_AZURE_AUTHORITY_HOST);
+        String clientId = CoreUtils.isNullOrEmpty(workloadIdentityClientId) ? managedIdentityClientId : workloadIdentityClientId;
+        if (!(CoreUtils.isNullOrEmpty(tenantId)
+            || CoreUtils.isNullOrEmpty(federatedTokenFilePath)
+            || CoreUtils.isNullOrEmpty(clientId)
+            || CoreUtils.isNullOrEmpty(azureAuthorityHost))) {
+            return new WorkloadIdentityCredential(clientId, tenantId, federatedTokenFilePath, identityClientOptions.setAuthorityHost(azureAuthorityHost).clone());
```

Comment:
```suggestion
            return new WorkloadIdentityCredential(tenantId, clientId, federatedTokenFilePath, identityClientOptions.setAuthorityHost(azureAuthorityHost).clone());
```
Should also have a test case that would catch this by validating the tenant and client IDs. The client ID especially needs test coverage because there's 3 ways to set it and all of them can be used simultaneously.

Reply:
added more tests to validate this.